### PR TITLE
change escape way  into encode

### PR DESCRIPTION
**Pull request by cheese8**
## Describe what this PR does for and how you did.
Ref #247

## Does this PR be associated with issue? If so, please adding the issue link below.
change escape & unescape way into encode & decode way

### Checklist
- [x] Code compiles correctly
- [ ] Pull Request has submit to 2020.0 and Greenwich
- [x] Create at least one junit test if possible
- [x] All tests passing
- [x] Extend documentation if necessary
- [x] Add myself / the copyright holder to the AUTHORS file

### CODE REVIEW

**Comment by SkyeBeFreeman** [comment]
I will review your PR later^ ^Plz request for reviewing when you have finished.

**Comment by cheese8** [comment]
> I will review your PR later^ ^Plz request for reviewing when you have finished.

yeah, thanks, it will be done. and next another pr is planed to remove escape, unscape and related useless code.

**Comment by cheese8** [comment]
@SkyeBeFreeman Can you help to review it？Local CI passed with  suppressed `ImportOrder`.

**Comment by SkyeBeFreeman** [comment]
> @SkyeBeFreeman It seems that `request someone for review` did not show on my page. Can you help to review it？Local CI passed with suppressed `ImportOrder`.

You can set checkstyle plugin in IDEA following what [代码规范](https://github.com/Tencent/spring-cloud-tencent/wiki/参与共建#代码规范) shows.

**Comment by cheese8** [comment]
> > @SkyeBeFreeman It seems that `request someone for review` did not show on my page. Can you help to review it？Local CI passed with suppressed `ImportOrder`.
> 
> You can set checkstyle plugin in IDEA following what [代码规范](https://github.com/Tencent/spring-cloud-tencent/wiki/%E5%8F%82%E4%B8%8E%E5%85%B1%E5%BB%BA#%E4%BB%A3%E7%A0%81%E8%A7%84%E8%8C%83) shows.

Thanks,  ready for review. @SkyeBeFreeman

**Comment by SkyeBeFreeman** [comment]
> > > @SkyeBeFreeman It seems that `request someone for review` did not show on my page. Can you help to review it？Local CI passed with suppressed `ImportOrder`.
> > 
> > 
> > You can set checkstyle plugin in IDEA following what [代码规范](https://github.com/Tencent/spring-cloud-tencent/wiki/%E5%8F%82%E4%B8%8E%E5%85%B1%E5%BB%BA#%E4%BB%A3%E7%A0%81%E8%A7%84%E8%8C%83) shows.
> 
> Thanks, ready for review. @SkyeBeFreeman

OK. I will review it tomorrow.

**Comment by SkyeBeFreeman** [comment]
@cheese8 I think you need delete `com.tencent.cloud.common.util.ExpressionLabelUtils#escape` and `com.tencent.cloud.common.util.ExpressionLabelUtils#unescape` for totally removing.

File: spring-cloud-starter-tencent-polaris-router/src/main/java/com/tencent/cloud/polaris/router/feign/PolarisFeignLoadBalancer.java [hunk 1]
```diff
@@ -70,18 +73,15 @@ PolarisRouterContext buildRouterContext(Map<String, Collection<String>> headers)
 		routerContext.setLabels(PolarisRouterContext.TRANSITIVE_LABELS, MetadataContextHolder.get()
 				.getFragmentContext(MetadataContext.FRAGMENT_TRANSITIVE));
 
-		labelHeaderValues.forEach(labelHeaderValue -> {
-			Map<String, String> labels = JacksonUtils.deserialize2Map(labelHeaderValue);
-			if (!CollectionUtils.isEmpty(labels)) {
-				Map<String, String> unescapeLabels = new HashMap<>(labels.size());
-				for (Map.Entry<String, String> entry : labels.entrySet()) {
-					String escapedKey = ExpressionLabelUtils.unescape(entry.getKey());
```
Comment (SkyeBeFreeman) [inline spring-cloud-starter-tencent-polaris-router/src/main/java/com/tencent/cloud/polaris/router/feign/PolarisFeignLoadBalancer.java:78]: delete `com.tencent.cloud.common.util.ExpressionLabelUtils#unescape` method.

File: spring-cloud-starter-tencent-polaris-router/src/main/java/com/tencent/cloud/polaris/router/feign/RouterLabelFeignInterceptor.java [hunk 2]
```diff
@@ -102,22 +105,20 @@ public void apply(RequestTemplate requestTemplate) {
 				.getFragmentContext(MetadataContext.FRAGMENT_TRANSITIVE);
 		labels.putAll(transitiveLabels);
 
-		// Because when the label is placed in RequestTemplate.header,
-		// RequestTemplate will parse the header according to the regular, which conflicts with the expression.
-		// Avoid conflicts by escaping.
-		Map<String, String> escapeLabels = new HashMap<>(labels.size());
-		for (Map.Entry<String, String> entry : labels.entrySet()) {
```
Comment (SkyeBeFreeman) [inline spring-cloud-starter-tencent-polaris-router/src/main/java/com/tencent/cloud/polaris/router/feign/RouterLabelFeignInterceptor.java:109]: delete `com.tencent.cloud.common.util.ExpressionLabelUtils#escape` method

**Review by SkyeBeFreeman — CHANGES_REQUESTED** [review]

File: spring-cloud-starter-tencent-polaris-router/src/main/java/com/tencent/cloud/polaris/router/feign/PolarisFeignLoadBalancer.java (hunk 1 above)
Comment (cheese8) [reply spring-cloud-starter-tencent-polaris-router/src/main/java/com/tencent/cloud/polaris/router/feign/PolarisFeignLoadBalancer.java:78]: In 2020.0 and Greenwich branch?

File: spring-cloud-starter-tencent-polaris-router/src/main/java/com/tencent/cloud/polaris/router/feign/RouterLabelFeignInterceptor.java (hunk 2 above)
Comment (cheese8) [reply spring-cloud-starter-tencent-polaris-router/src/main/java/com/tencent/cloud/polaris/router/feign/RouterLabelFeignInterceptor.java:109]: In 2020.0 or Greenwich branch? Can it be reviewed in main branch, if it's ok, merge it into 2020.0 or Greenwich?

**Comment by cheese8** [comment]
> next another pr is planed to remove escape, unscape and related useless code.

Yeah, do you think we could take it down by 3 steps？
1. change the escape & unescape into encode & decode
2. remove escape, unscape and related useless code.
3. merge or cherry-pick those into another branch.

**Comment by cheese8** [comment]
> @cheese8 I think you need delete `com.tencent.cloud.common.util.ExpressionLabelUtils#escape` and `com.tencent.cloud.common.util.ExpressionLabelUtils#unescape` for totally removing.

Yeah, do you think we could take it down by 3 steps？

change the escape & unescape into encode & decode
remove escape, unscape and related useless code.
merge or cherry-pick those into another branch.

**Review by SkyeBeFreeman — APPROVED** [review]
LGTM.

**Pull request merged**